addons/binding/org.openhab.binding.isy/src/main/java/org/openhab/binding/isy/handler/IsyHandlerBuilder.java

```diff
@@ -1,13 +1,20 @@
 package org.openhab.binding.isy.handler;
 
 import org.eclipse.smarthome.core.thing.Thing;
+import org.openhab.binding.isy.IsyBindingConstants;
+import org.openhab.binding.isy.handler.special.VenstarThermostatDeviceHandler;
 
 public class IsyHandlerBuilder {
 
     IsyDeviceHandler handler;
 
     protected IsyHandlerBuilder(Thing thing) {
-        this.handler = new IsyDeviceHandler(thing);
+        // for thermostat, return a specialized handler based on type UID
+        if (IsyBindingConstants.VENSTAR_THERMOSTAT_THING_TYPE.equals(thing.getThingTypeUID())) {
```

Comment:
I think it would be better to not add the thermo support to this builder.  I created this builder so we could use a generic handler for most of the isy items.  However thermostats are quite the beast on their own, and you underestandably you created the specific handler.  In the handler factory I would just instantiate the handler you wrote, and then you could embed all that channel mapping right into the handler class, instead of creating it and passing in to the builder.  Make sense?

Reply:
Yea, I wasn't too happy about this choice myself - it was basically the fastest way to get that handler created. I'll change it to what you say here, just give me a couple days.